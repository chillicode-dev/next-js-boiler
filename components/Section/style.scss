.Section {
  padding: 0;
}

.hasPaddings {
  padding: 60px 0;
}

.transparent {
  background-color: transparent;
}

.bg {
  background-color: $bgColor;
}

.brand {
  background-color: $brandColor;
  color: white;
}
