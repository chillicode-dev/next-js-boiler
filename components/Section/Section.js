// Vendor
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
// Internals
import Container from 'components/Container';
import Typography from 'components/Typography';
import style from './style.scss';

const Section = ({className, children, title, hasPaddings, color}) => {
  const classes = cn({
    [style.Section]: true,
    [style.hasPaddings]: hasPaddings,
    [style[color]]: color,
    [className]: className,
  });

  return (
    <div className={classes}>
      <Container>
        {title && (
          <Typography variant="heading2" hasMargin>
            {title}
          </Typography>
        )}
        <div>{children}</div>
      </Container>
    </div>
  );
};

Section.defaultProps = {
  className: '',
  title: '',
  hasPaddings: true,
  color: 'transparent',
};

Section.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  hasPaddings: PropTypes.bool,
  color: PropTypes.oneOf(['transparent', 'bg', 'brand']),
};

export default Section;
