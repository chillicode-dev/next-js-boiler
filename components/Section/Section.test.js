import React from 'react';
import renderer from 'react-test-renderer';
import Section from './Section';

it('Renders Section correctly', () => {
  const elements = (
    <div>
      <Section>Text</Section>
      <Section title="Section 1">Text</Section>
      <Section title="Section 2" color="brand">
        Text
      </Section>
      <Section title="Section without paddings" hasPaddings={false} color="bg">
        Text
      </Section>
    </div>
  );
  const tree = renderer.create(elements).toJSON();
  expect(tree).toMatchSnapshot();
});
