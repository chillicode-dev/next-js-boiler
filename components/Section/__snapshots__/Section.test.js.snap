// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Section correctly 1`] = `
<div>
  <div
    className="Section hasPaddings transparent"
  >
    <div
      className="Container"
    >
      
      <div>
        Text
      </div>
    </div>
  </div>
  <div
    className="Section hasPaddings transparent"
  >
    <div
      className="Container"
    >
      <h2
        className="Typography heading2 hasMargin"
      >
        Section 1
      </h2>
      <div>
        Text
      </div>
    </div>
  </div>
  <div
    className="Section hasPaddings brand"
  >
    <div
      className="Container"
    >
      <h2
        className="Typography heading2 hasMargin"
      >
        Section 2
      </h2>
      <div>
        Text
      </div>
    </div>
  </div>
  <div
    className="Section bg"
  >
    <div
      className="Container"
    >
      <h2
        className="Typography heading2 hasMargin"
      >
        Section without paddings
      </h2>
      <div>
        Text
      </div>
    </div>
  </div>
</div>
`;
